internal/api/v1/user/transactions/transactions_api_test.go

```diff
@@ -15,6 +15,102 @@ import (
 	"github.com/stretchr/testify/require"
 )
 
+func TestTransactionsAPI_SendToRecipients(t *testing.T) {
+	tests := map[string]struct {
+		draftTransactionResponder  httpmock.Responder
+		recordTransactionResponder httpmock.Responder
+		expectedResponse           *response.Transaction
+		expectedErr                error
+	}{
+		"HTTP POST /api/v1/transactions/drafts & /api/v1/transactions response: 200": {
+			expectedResponse:           transactionstest.ExpectedSendToRecipientsTransaction(t),
+			recordTransactionResponder: httpmock.NewJsonResponderOrPanic(http.StatusOK, httpmock.File("transactionstest/transaction_send_to_recipients_200.json")),
+			draftTransactionResponder:  httpmock.NewJsonResponderOrPanic(http.StatusOK, httpmock.File("transactionstest/transaction_draft_with_hex_200.json")),
+		},
+		"HTTP POST /api/v1/transactions/drafts & /api/v1/transactions response: 400": {
+			expectedErr:                transactionstest.NewBadRequestSPVError(),
+			draftTransactionResponder:  httpmock.NewJsonResponderOrPanic(http.StatusBadRequest, transactionstest.NewBadRequestSPVError()),
+			recordTransactionResponder: httpmock.NewJsonResponderOrPanic(http.StatusBadRequest, transactionstest.NewBadRequestSPVError()),
+		},
+		"HTTP POST /api/v1/transactions/drafts & /api/v1/transactions response: 500": {
+			expectedErr:                transactionstest.NewInternalServerSPVError(),
+			draftTransactionResponder:  httpmock.NewJsonResponderOrPanic(http.StatusInternalServerError, transactionstest.NewInternalServerSPVError()),
+			recordTransactionResponder: httpmock.NewJsonResponderOrPanic(http.StatusInternalServerError, transactionstest.NewInternalServerSPVError()),
+		},
+	}
```

Comment:
```
func (a *API) SendToRecipients(ctx context.Context, r *commands.SendToRecipients) (*response.Transaction, error) {
	draft, err := a.DraftToRecipients(ctx, r)
	if err != nil {
		return nil, err
	}

	var hex string
	if hex, err = a.FinalizeTransaction(draft); err != nil {
		return nil, err
	}

	recordTransactionCmd := &commands.RecordTransaction{
		Metadata:    r.Metadata,
		Hex:         hex,
		ReferenceID: draft.ID,
	}
	return a.RecordTransaction(ctx, recordTransactionCmd)
}
```
After the first failure caused by the _DraftToRecipients_ method call, the method should terminate immediately and return the appropriate error. The test structure can be adjusted so that unnecessary responders aren’t set.

Based on this logic flow, I would suggest splitting the test into separate cases that cover the following scenarios:

1. All three calls were successful (_DraftToRecipients_, _FinalizeTransaction_, _RecordTransaction_).
1. Only _DraftToRecipients_ was successful.
1. Both _DraftToRecipients_ and _FinalizeTransaction_ were successful.

This approach would make the tests clearer and more focused. 😸